src/routes/revenue.route.ts

```diff
@@ -1,20 +1,28 @@
 import express from 'express';
 import useCatchErrors from '../error/catchErrors';
 import RevenueController from '../controller/revenue.controller';
+import { isAuthenticated } from '../middlewares/auth';
 
 export default class RevenueRoute {
   router = express.Router();
   revenueController = new RevenueController();
   path = '/revenues';
 
-  //   constructor() {
-  //     this.initializeRoutes();
-  //   }
+  constructor() {
+    this.initializeRoutes();
+  }
 
   initializeRoutes() {
     this.router.patch(
       `${this.path}/:order_id`,
-      useCatchErrors(this.revenueController.updateRevenue.bind(this.revenueController)),
+      useCatchErrors(this.revenueController.updateRevenue.bind(this.revenueController))
+    );
+
+    // Route for /api/revenues?timeframe=today
+    this.router.get(
```

Comment:
why did you change the base `path` when you can easily add s at the end of base path`${this.path}s` ? please revert it back as all other endpoint without an `s` depends on it.